sycl/test/functor/functor_inheritance.cpp

```diff
@@ -1,4 +1,4 @@
-// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -o %t.out %s
```

Comment:
This test does not need -fsycl-unnamed-lambda to pass. @erichkeane , why did you change this one?

Reply:
Same question to kernel_functor.cpp, which probably also does not need -fsycl-unnamged-lambda option as well.
(but I did not try compiling that test with your changes).

Reply:
Because it used to :)  This patch unfortunately has grown a whole bunch as a result of seeing all the issues, so the flag wasn't 'default' when I first wrote this patch.  I'll audit/remove uses of it.